### ISSUE TITLE
Report "Immediately trying to open a thread you made from catalog results in 404"

### ISSUE THREAD

**Issue by K1rakishou** [Adamantcheese/Kuroba#915]
UUID: 4faf7296ffa6e4f4660acaa5a78f5964
Build flavor: dev
Version name: v4.12.0-b1dc523
OS info: Android 10, sdk version: 29
Report description: Problem as described. I created a thread on /g/, and I tried opening it from the catalog and it gave me the 404 page. I was able to open it from the bookmark side bar though. Restarting also fixed the issue.

**Comment by Astridchan** [comment]
I can confirm this bug (or at least I'm fairly certain it's the same bug). Doesn't need to be a thread you've created, it can happen with any thread that's new enough.

Video of the bug in action:
https://streamable.com/y0mai6

Force closing the app fixes it but after a while it will start happening again.

**Comment by Adamantcheese** [comment]
Yeah it's weird, I had it happen on emulator once, but then restarting the emulator made the failure go away and I can't reproduce it anymore. Tried posting a thread and that went fine, opening up all sorts of threads went fine until it didn't. Like the report says, the response code from OkHttp was a 404, which doesn't make sense because the thread definitely was still alive.

**Comment by Adamantcheese** [comment]
Also of note: this was before any of the OkHttp migrations were added in for ChanThreadLoader, and given that it still occurred afterwards it is likely not an issue with any of the commits after the one listed in the version.

**Comment by Astridchan** [comment]
One thing I noticed about this bug is if you get an active thread that's showing up as 404 in Kuroba, when you try to open that thread through the archive links in app the post number it uses is completely different, eg. when I try to open https://boards.4chan.org/g/thread/76248465 in Kuroba it shows the thread is 404 and when I click to open the thread in an archive site it takes me to https://archive.rebeccablacktech.com/g/thread/76248403. Even weirder is that other threads that are also showing wrongly as 404 on Kuroba also open the same URL with the same post number when you try to open the thread in the archives.

EDIT: seems that the post number in the archive link URL is always a post number of a reply I've made in other threads. https://archive.rebeccablacktech.com/g/thread/76226366/#76248403 is my post.

**Comment by Adamantcheese** [comment]
The thread title also says `Title will update shortly…`, correct? I think I remember that's what it said, which points me to it being an issue with threads after you post something, i.e. without posting anything this issue won't occur, but will occur afterwards. I have reason to believe it was due to the removal of the "threadNo" field in ReplyResponse in https://github.com/Adamantcheese/Kuroba/commit/c3b2c4549ee47a1629406a23816563c7973bcfa5, which I can revert the changes that likely caused the problem. This probably is also the cause of #912. I've pushed that change, and if it no longer occurs, great! Otherwise there's more testing to be done.
